docs/src/models/stokes_drift.md

```diff
@@ -0,0 +1,209 @@
+uniformly across the domain.
+
+To construct a [`UniformStokesDrift`](@ref), you must provide functions for the vertical derivatives
+of the Stokes drift velocity components. The required derivatives are `∂z_uˢ` and `∂z_vˢ`. If the
+Stokes drift is time-dependent, you also provide `∂t_uˢ` and `∂t_vˢ`.
+
+### Example: Exponentially-decaying Stokes drift
+
+The Stokes drift from a monochromatic deep-water surface wave decays exponentially with depth.
+For a wave with wavelength ``\lambda`` propagating in the ``x``-direction, the Stokes drift is
+
+```math
+u^S(z) = U^S \exp(2 k z)
+```
+
+where ``k = 2\pi / \lambda`` is the wavenumber and ``U^S`` is the surface Stokes drift velocity.
+The vertical derivative is ``\partial_z u^S = 2 k U^S \exp(2 k z)``.
+
+```jldoctest stokes
+using Oceananigans
+
+wavelength = 60  # meters
+wavenumber = 2π / wavelength
+Uˢ = 0.01  # m/s, surface Stokes drift
+∂z_uˢ(z, t) = 2 * wavenumber * Uˢ * exp(2 * wavenumber * z)
+stokes_drift = UniformStokesDrift(∂z_uˢ = ∂z_uˢ)
+
+# output
+
+UniformStokesDrift{Nothing}:
+├── ∂z_uˢ: ∂z_uˢ
+├── ∂z_vˢ: zerofunction
+├── ∂t_uˢ: zerofunction
+└── ∂t_vˢ: zerofunction
+```
+
+### Example: Using parameters
+
+Instead of defining global constants, you can pass a `parameters` NamedTuple. When `parameters` is
+provided, the Stokes drift functions are called with the signature `(z, t, parameters)`:
+
+```jldoctest stokes
+∂z_uˢ(z, t, p) = 2 * p.k * p.Uˢ * exp(2 * p.k * z)
+stokes_params = (k = 2π / 60, Uˢ = 0.01)
+stokes_drift = UniformStokesDrift(∂z_uˢ = ∂z_uˢ, parameters = stokes_params)
+
+# output
+
+UniformStokesDrift with parameters (k=0.10472, Uˢ=0.01):
+├── ∂z_uˢ: ∂z_uˢ
+├── ∂z_vˢ: zerofunction
+├── ∂t_uˢ: zerofunction
+└── ∂t_vˢ: zerofunction
+```
+
+### Using `UniformStokesDrift` in a model
+
+To include Stokes drift in a [`NonhydrostaticModel`](@ref), pass the `stokes_drift` object as a
+keyword argument:
+
+```jldoctest stokes
+grid = RectilinearGrid(size = (32, 32, 32), extent = (128, 128, 64))
+∂z_uˢ(z, t) = 0.01 * exp(z / 10)
+model = NonhydrostaticModel(; grid, stokes_drift = UniformStokesDrift(∂z_uˢ = ∂z_uˢ))
+model.stokes_drift
+
+# output
+
+UniformStokesDrift{Nothing}:
+├── ∂z_uˢ: ∂z_uˢ
+├── ∂z_vˢ: zerofunction
+├── ∂t_uˢ: zerofunction
+└── ∂t_vˢ: zerofunction
+```
+
+## `StokesDrift`
+
+[`StokesDrift`](@ref) is used for three-dimensional, spatially-varying Stokes drift fields such
+as might arise from a wave packet with a horizontally-varying envelope:
+
+```math
+\boldsymbol{u}^S = u^S(x, y, z, t) \hat{\boldsymbol{x}} + v^S(x, y, z, t) \hat{\boldsymbol{y}} + w^S(x, y, z, t) \hat{\boldsymbol{z}}
+```
+
+!!! warning "Divergence-free requirement"
+    For three-dimensional Stokes drift, the provided Stokes drift field must be divergence-free
+    (solenoidal): ``\nabla \cdot \boldsymbol{u}^S = 0``. This is required for consistency with
+    the Craik-Leibovich formulation as discussed by [Vanneste and Young (2022)](@cite VannesteYoung2022).
```

Comment:
Since it's not within a docstring and we'd never be wanting to see this via the REPL, we can just use the simpler `@citet`:

```suggestion
    the Craik-Leibovich formulation as discussed by [VannesteYoung2022](@citet).
```